akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -83,6 +81,8 @@ private[remote] object HeaderBuilder {
 
   def out(): HeaderBuilder =
     new HeaderBuilderImpl(NoInboundCompressions, CompressionTable.empty[ActorRef], CompressionTable.empty[String])
+
+  final val DeadLettersCode = -1
```

Comment:
population of the table starts at 0 (in DecompressionTable it's an array), and previously we pre-populated the table with deadletters, but I don't think that was completely correct and we don't need it in the actual table